sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -80,10 +80,31 @@
      *     operations on the message broker.
      * @param retryOptions for the session operations.
      */
+    public ReactorSession(Session session, SessionHandler sessionHandler, String sessionName, ReactorProvider provider,
```

Comment:
Do we need another ctor overload for this? I think this is overkill because only we use this. Can we bag this into CreateSessionOptions that only we use?

Reply:
I thought about that, but did not wanted to introduce another class `CreateSessionOptions` but it will help us in future to add more properties to it. So I will add `CreateSessionOptions`.

Reply:
Looks the same still?

Reply:
I added ` CreateSessionOptions` based on first comment.

Reply:
I meant to include parameters like MessageSerializer and AmqpRetryOptions so we don't need two new overloads.

Reply:
More generally, does this transaction logic _need_ to be in here? It is very specific to service bus and we already extend from the class in SB.  Can you use a combination of the existing methods to perform this?